src/platform_impl/macos/view.rs

```diff
@@ -1126,3 +1147,26 @@ fn replace_event(event: &NSEvent, option_as_alt: OptionAsAlt) -> Id<NSEvent> {
         event.copy()
     }
 }
+
+// Convert given NSRange to Rust String Range data used in this crate.
```

Comment:
From just reading the code, I cannot tell what this code does, maybe describe it more in words, that it converts from the UTF-16 that AppKit uses in its NSRange to the UTF-8 that Rust uses.

And why is it marked as `unsafe`? Maybe it'll be more clear if inlined instead of split into a separate function?

Reply:
@madsmtm Thanks. Pan93412 also has the same opinion. I just fixed this.